bin/diagnostic.js

```diff
@@ -1 +1,15 @@
 'use strict';
+const ask = require('../lib/ask.js');
+
+let answer = 'not empty';
+
+while (answer !== '') {
+  answer = ask("Guess my secret? ");
+  if (answer < 0 || answer > 10) {
+    console.log('You guess needs to be between 1 and 10')
+  }
+  if (answer === '4') {
```

Comment:
Your code works perfectly. However, the solution can be optimized by using just 1 if-statement instead of 2. Multiple if-statements come in handy when you are assessing properties that are independent of each other. Since both of these are testing the value of a, you could just use an "else" clause.